src/core/catalog.js

```diff
@@ -1710,6 +1722,15 @@ class ExtendedCatalog extends Catalog {
     return tree !== null && isDict(tree) && tree.has('RoleMap') ? tree.get('RoleMap') : new Map();
   }
 
+  getRoleName(el, name) {
+    let namespace = isDict(el) && el.has('NS') ? el.get('NS') : null;
+    let roleNameNS = isDict(namespace) && namespace.has('RoleMapNS') ? namespace.get('RoleMapNS') : null;
+    let roleNameNSArray = isDict(roleNameNS) && roleNameNS.has(name) ? roleNameNS.get(name) : null;
+    let roleName_v1 = this.roleMap.get(name) !== undefined ? this.roleMap.get(name).name : null;
```

Comment:
Why not just to check for a falsy value? `this.roleMap.get(name) ? this.roleMap.get(name).name : null;`